wgsl/index.bs

```diff
@@ -6973,6 +6999,15 @@ textureSample(t : texture_depth_cube_array, s : sampler, coords : vec3<f32>, arr
 
 The sampled value.
 
+### `textureSampleLuminance` ### {#texturesampleluminance}
+
+Samples a luminance value from a `texture_external`.
+Must only be used in a [=fragment=] shader stage.
```

Comment:
Is there a reason why this must only be used in fragment?

Reply:
I copied this restriction from the section on `textureSample`.

Reply:
Actually, we have a real use case that requires this `textureSample` works in `compute stage` too.

Reply:
@shaoboyan Can that use case use `textureLoad` instead? I think there are good reasons `textureSample` is only allowed in fragment shaders (though I don't know the details).

Reply:
@kainino0x Check the shader logic and yes, the `textureLoad` could be used instead.

Reply:
Or `textureSampleLevel`. The problem with `textureSample` is that it uses an implicit LOD. Here external textures have a single LOD so it is fully explicit (and is 0). So I think we could remove this restriction and provide `textureSampleLuminance` and `textureSample` of `external_texture` in all stages.

That details (and whether to call these `textureSampleNoLod` or something else) are probably worth bringing to the WGSL group.

Reply:
Do we need to add textureSampleLevel for external_texture then?